### PR TITLE
CY-1469 - Create repository with reusable React components (p. 1)

### PR DESCRIPTION
**Pull request by qooban**
# Changelog
- Setup development environment for React components (CY-1482)
- Added documentation support - StoryBook Docs (CY-1483)
- Added storybook support for new components (CY-1484)

# Screenshot
image

### CODE REVIEW

File: test/Checkmark.test.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import Checkmark from '../src/components/Checkmark/Checkmark';
+
+describe('Checkmark', () => {
+    let wrapper = mount(<Checkmark value />);
+
+    test('renders properly', () => {
```
Comment (tyacbovi) [inline test/Checkmark.test.jsx]: what about checking it's readonly?

File: test/__snapshots__/StoryShots.test.js.snap [hunk 2]
```diff
@@ -0,0 +1,70 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (tyacbovi) [inline test/__snapshots__/StoryShots.test.js.snap]: Do we need this file?

File: .storybook/theme.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import { create } from '@storybook/theming';
+import 'semantic-ui-css/semantic.min.css';
+
+export default create({
+    base: 'light',
+
+    // colorPrimary: 'hotpink',
```
Comment (tyacbovi) [inline .storybook/theme.js]: Is this by intent? the comment out code?

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+      - run:
+          name: Install dependencies
+          command: npm ci
+
+      - run:
+          name: Build application
+          command: npm run build
+
+      - run:
```
Comment (tyacbovi) [inline .circleci/config.yml]: maybe move the lint job to a separate job, so it will happen in parallel

File: test/Checkmark.test.jsx (hunk 1 above)
Comment (qooban) [reply test/Checkmark.test.jsx]: Tests are just migrated from Stage right now. We can think of extending them, but I didn't focus on that now.

File: test/__snapshots__/StoryShots.test.js.snap (hunk 2 above)
Comment (qooban) [reply test/__snapshots__/StoryShots.test.js.snap]: Yes, it was created by snapshot testing framework and is used in snapshot testing.

File: .storybook/theme.js (hunk 3 above)
Comment (qooban) [reply .storybook/theme.js]: Yes, I plan to do the theming in one of the next steps. Just created sub-task for that (https://cloudifysource.atlassian.net/browse/CY-1826).

File: .circleci/config.yml (hunk 4 above)
Comment (qooban) [reply .circleci/config.yml]: Not sure, right now lint job takes ~5s:
```
jni@jni-PC MINGW64 ~/WebstormProjects/cloudify-ui-components (CY-1469)
$ time npm run lint

> cloudify-ui-components@1.0.0-alpha.1 lint C:\Users\jni\WebstormProjects\cloudify-ui-components
> eslint --ext js,jsx src


real    0m5.193s
user    0m0.046s
sys     0m0.606s

```
so I don't think we will benefit from the parallel execution.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  "version": "1.0.0-alpha.1",
+  "description": "Cloudify UI React components library",
+  "homepage": "https://github.com/cloudify-cosmo/cloudify-ui-components#readme",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/cloudify-cosmo/cloudify-ui-components.git"
+  },
+  "license": "Apache-2.0",
+  "author": {
+    "name": "Jakub Niezgoda",
```
[the remark this answers is not in the document]
Comment (qooban) [reply package.json]: I used my Cloudify company mail here the same as in `cloudify-ui-common\package.json`.
Which other e-mail do you suggest? What do you write in backend repositories?

When I leave Cloudify I assume that `jakub.niezgoda@cloudify.co` will be redirected to some generic mailbox or dropped.

[the remark this answers is not in the document]
Comment (qooban) [reply package.json]: Created https://cloudifysource.atlassian.net/browse/CY-1828 to align that.

**Pull request merged**